src/python/pants/reporting/reporting.py

```diff
@@ -98,6 +101,11 @@ def initialize(self, run_tracker, all_options, start_time=None):
     html_reporter = HtmlReporter(run_tracker, html_reporter_settings)
     report.add_reporter('html', html_reporter)
 
+    # Set up JSON reporting.
+    json_reporter_settings = JsonReporter.Settings(log_level=Report.INFO, json_dir=json_dir)
+    json_reporter = JsonReporter(run_tracker, json_reporter_settings)
```

Comment:
The `run_tracker` here is required by the `Reporter` base class, where it's only used for determining if the reporter is under the main root or not. The only thing that the json reporter uses it for is for artifact cache stats (this first pass assumes the flag would be either the current stats _or_ the `JsonReporter` stats - which is why the cache stats were included).

Perhaps the run tracker could continue to hold the top-level stats (as is done currently), and simply delegate detailed workunit stats to this reporter behind an option.